lang/java/reef-examples/src/main/java/org/apache/reef/examples/data/output/OutputServiceDriver.java

```diff
@@ -0,0 +1,138 @@
+import org.apache.reef.io.data.output.OutputService;
+import org.apache.reef.io.data.output.OutputStreamProvider;
+import org.apache.reef.io.data.output.OutputStreamProviderHDFS;
+import org.apache.reef.io.data.output.OutputStreamProviderLocal;
+import org.apache.reef.tang.Configuration;
+import org.apache.reef.tang.annotations.Parameter;
+import org.apache.reef.tang.annotations.Unit;
+import org.apache.reef.wake.EventHandler;
+import org.apache.reef.wake.time.event.StartTime;
+
+import javax.inject.Inject;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * The Driver code for the output service demo app.
+ */
+@Unit
+public final class OutputServiceDriver {
+  private static final Logger LOG = Logger.getLogger(OutputServiceDriver.class.getName());
+
+  /**
+   * evaluator requestor object used to create new evaluator containers.
+   */
+  private final EvaluatorRequestor requestor;
+
+  /**
+   * true for local runtime, or false for YARN runtime.
+   */
+  private final Boolean isLocal;
+
+  /**
+   * path of the output directory.
+   */
+  private final String outputDir;
+
+  /**
+   * Sub-id for Tasks.
+   * This object grants different IDs to each task
+   * e.g. Task-0, Task-1, and so on.
+   */
+  private final AtomicInteger taskId = new AtomicInteger(0);
+
+  /**
+   * Job driver constructor - instantiated via TANG.
+   *
+   * @param requestor evaluator requestor object used to create new evaluator containers.
+   * @param isLocal true for local runtime, or false for YARN runtime.
+   * @param outputDir path of the output directory.
+   */
+  @Inject
+  public OutputServiceDriver(
+      final EvaluatorRequestor requestor,
+      @Parameter(OutputServiceREEF.Local.class) final Boolean isLocal,
+      @Parameter(OutputServiceREEF.OutputDir.class) final String outputDir) {
+    LOG.log(Level.FINE, "Instantiated 'OutputServiceDriver'");
+    this.requestor = requestor;
+    this.isLocal = isLocal;
+    this.outputDir = outputDir;
+  }
+
+  /**
+   * Handles the StartTime event: Request three Evaluators.
+   */
+  public final class StartHandler implements EventHandler<StartTime> {
+    @Override
+    public void onNext(final StartTime startTime) {
+      OutputServiceDriver.this.requestor.submit(EvaluatorRequest.newBuilder()
+          .setNumber(3)
+          .setMemory(64)
+          .setNumberOfCores(1)
+          .build());
+      LOG.log(Level.INFO, "Requested Evaluator.");
+    }
+  }
+
+  /**
+   * Handles AllocatedEvaluator: Submit the output service and a context for it.
+   */
+  public final class EvaluatorAllocatedHandler implements EventHandler<AllocatedEvaluator> {
+    @Override
+    public void onNext(final AllocatedEvaluator allocatedEvaluator) {
+      LOG.log(Level.INFO, "Submitting Output Service to AllocatedEvaluator: {0}", allocatedEvaluator);
+      final Class<? extends OutputStreamProvider> outputStreamProviderClass =
```

Comment:
Yes, that could be an object method, and an instance of `OutputService` is injected into the Drivers that use it.